toplevel/vernac.ml

```diff
@@ -81,14 +81,15 @@ let interp_vernac ~check ~interactive ~state ({CAst.loc;_} as com) =
       Exninfo.iraise (reraise, info)
 
 (* Load a vernac file. CErrors are annotated with file and location *)
-let load_vernac_core ~echo ~check ~interactive ~state file =
+let load_vernac_core ~echo ~check ~interactive ~state ?ldir file =
   (* Keep in sync *)
   let in_chan = open_utf8_file_in file in
   let in_echo = if echo then Some (open_utf8_file_in file) else None in
   let input_cleanup () = close_in in_chan; Option.iter close_in in_echo in
 
-  let in_pa =
-    Pcoq.Parsable.make ~loc:(Loc.(initial (InFile file)))
+  let moduel = Option.cata (fun ldir -> Some Names.DirPath.(to_string ldir))
+      None ldir in
+  let in_pa = Pcoq.Parsable.make ~loc:Loc.(initial (InFile (moduel, file)))
```

Comment:
Spelling
```suggestion
  let module = Option.cata (fun ldir -> Some Names.DirPath.(to_string ldir))
      None ldir in
  let in_pa = Pcoq.Parsable.make ~loc:Loc.(initial (InFile (module, file)))
```

Reply:
`module` is a reserved word, so I can't use that.  What would you suggest?

Reply:
coq_module, maybe?  module_name?

Reply:
Made it dirpath.